packages/aws-cdk-lib/core/test/helpers-internal/validate-props.test.ts

```diff
@@ -0,0 +1,90 @@
+import { Construct } from 'constructs';
+import { ValidationError } from '../../lib/errors';
+import { validateProps, ValidationRule } from '../../lib/helpers-internal/validate-props';
+
+class MockConstruct extends Construct {
```

Comment:
💅🏻 nit: In the context of jest tests "mock" means a very specific thing, I usually call this `Fake` or `Test` just to not confuse a future reader of the code.

Reply:
Will rename to TestConstruct.